kernel/interrupt_controller/src/aarch64.rs

```diff
@@ -111,126 +112,92 @@ impl SystemInterruptControllerApi for SystemInterruptController {
         priority: Priority,
     ) -> Result<(), &'static str> {
         assert!(sys_int_num >= 32, "shared peripheral interrupts have a number >= 32");
-        let mut int_ctlr = INTERRUPT_CONTROLLER.lock();
-        let int_ctlr = int_ctlr
-            .as_mut()
-            .expect("BUG: set_destination(): INTERRUPT_CONTROLLER was uninitialized");
+        let mut dist = self.0.lock();
 
-        int_ctlr.set_spi_target(sys_int_num as _, SpiDestination::Specific(destination));
-        int_ctlr.set_interrupt_priority(sys_int_num as _, priority);
+        dist.set_spi_target(sys_int_num as _, SpiDestination::Specific(destination));
+        dist.set_spi_priority(sys_int_num as _, priority);
 
         Ok(())
     }
 }
 
 impl LocalInterruptControllerApi for LocalInterruptController {
-    fn init_secondary_cpu_interface(&self) {
-        let mut int_ctlr = INTERRUPT_CONTROLLER.lock();
-        let int_ctlr = int_ctlr
-            .as_mut()
-            .expect("BUG: init_secondary_cpu_interface(): INTERRUPT_CONTROLLER was uninitialized");
+    fn get() -> &'static Self {
+        // While we're waiting for cpu-local-storage, this loop will work as fine as an AtomicMap
 
-        int_ctlr.init_secondary_cpu_interface();
+        let cpu_id = current_cpu();
+        let index = BOARD_CONFIG.cpu_ids.iter().position(|mpidr| cpu_id == (*mpidr).into());
+        let index = index.expect("Invalid CpuId returned by current_cpu()");
+
+        let ctrls = LOCAL_INT_CTRL.get();
+        let ctrls = ctrls.expect("interrupt_controller wasn't initialized");
```

Comment:
this series of operations is confusing. You can clarify it by describing the order of indexing used in the `LOCAL_INT_CTRL` (on the docs for that static var), and how it is explicitly _not_ directly indexable by the CPU ID value itself.